googleapiclient/discovery_cache/file_cache.py

```diff
@@ -0,0 +1,89 @@
+# Copyright 2014 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""File based cache for the discovery document."""
+
+
+import datetime
+import json
+import logging
+import os
+import tempfile
+import threading
+
+from . import base
+
+
+FILENAME = 'discovery-doc.cache'
+EPOCH = datetime.datetime.utcfromtimestamp(0)
+
+
+def to_timestamp(d):
+  return (d - EPOCH).total_seconds()
+
+
+class Cache(base.CacheBase):
+  """A file based cache for the discovery documents.
+
+  Attributes:
```

Comment:
Why are these instance-scope attributes instance-public? What's the use case for the lock for a user outside of this class?